### PR TITLE
Oasis 2: A Real Town

### PR DESCRIPTION
**Pull request by John-Dang**
## About The Pull Request
image
image
image
I have realigned the Oasis Z-levels, and majorly redone Oasis. The deputies and sheriff have been edited, along with the shopkeeper, barkeep and mayor. Names have been changed, loadouts, and Oasis has been almost entirely remade from scratch.
image
Also, I made some changes to the arrivals area at Central Command, so you're showing up at someplace that exists in the universe, and isn't Central Command from Space Station 13. Its a train station stop that's been converted by raiders!

## Why It's Good For The Game
Oasis has been reborn. The new mapping encourages using the whole buffalo of the town interior, encourages the followers to integrate with the community more and not just turtle in their clinic, town congregation, the mayor selling properties, private residences, a nicer innhouse, an inn instead of a bar, a proper Oasis Police Department, and more! It makes Oasis much better!

## Changelog
:cl:
add: Added Oasis Uptown
add: Added Oasis Undertown
add: New bumpersword description!
del: Removed old Oasis
tweak: Some poster and item descriptions have been changed.
tweak: Most town jobs have new descriptions and supervisors assigned to them.
balance: Changed specific weapon tiers to random or random high in Oasis, lowered merchant's material count.
/:cl:

### CODE REVIEW

**Comment by AdamElTablawy** [comment]
Be specific. What did you give to each thing?

File: code/modules/jobs/job_types/den.dm [hunk 1]
```diff
@@ -53,10 +53,10 @@ Mayor
 	l_pocket = /obj/item/storage/bag/money/small/settler
 	r_pocket = /obj/item/flashlight/flare
 	belt = /obj/item/gun/ballistic/revolver/colt357
-	shoes = 		/obj/item/clothing/shoes/f13/tan
-	uniform = /obj/item/clothing/under/f13/gentlesuit
-	suit = /obj/item/clothing/suit/armor/f13/town/mayor
-	head = /obj/item/clothing/head/f13/town/mayor
+	shoes = 		/obj/item/clothing/shoes/laceup
+	uniform = /obj/item/clothing/under/suit/checkered
+	suit = /obj/item/clothing/suit/armor/f13/kit
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/den.dm:58]: Why? Why doesn't he get the mayor hat and coat anymore?

File: code/modules/jobs/job_types/den.dm [hunk 2]
```diff
@@ -93,11 +93,11 @@ Mayor
...
 
 	ears = 			/obj/item/radio/headset/headset_town
-	uniform =  		/obj/item/clothing/under/f13/sheriff
+	uniform =  		/obj/item/clothing/under/f13/police
 	neck =			/obj/item/storage/belt/holster
-	shoes = 		/obj/item/clothing/shoes/f13/cowboy
-	suit = 			/obj/item/clothing/suit/armor/f13/town/sheriff
-	head = 			/obj/item/clothing/head/f13/town/sheriff
+	shoes = 		/obj/item/clothing/shoes/combat/swat
+	suit = 			/obj/item/clothing/suit/armor/vest/warden
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/den.dm:99]: Why does the sherrif lose their hat and coat?

File: code/modules/jobs/job_types/den.dm [hunk 3]
```diff
@@ -174,8 +174,8 @@ Mayor
 	flag = F13FARMER
 	department_flag = DEN
 	faction = "Town"
-	total_positions = 4
-	spawn_positions = 4
+	total_positions = 0
+	spawn_positions = 0
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/den.dm:178]: Why isn't farmer a role anymore?

File: code/modules/jobs/job_types/den.dm [hunk 4]
```diff
@@ -220,8 +220,8 @@ Mayor
 	flag = F13PROSPECTOR
 	department_flag = DEN
 	faction = "Town"
-	total_positions = 4
-	spawn_positions = 4
+	total_positions = 0
+	spawn_positions = 0
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/den.dm:224]: Why isn't prospector a role anymore?

File: code/modules/jobs/job_types/den.dm [hunk 5]
```diff
@@ -753,10 +753,10 @@ Mayor
 	flag = F13SHOPKEEPER
 	department_flag = DEN
 	faction = "Town"
-	total_positions = 2
-	spawn_positions = 2
-	supervisors = "law and order"
-	description = "The capitalist economy of pre-war america survived alongside its people. Now it's your job to continue its survival, so make some caps!"
+	total_positions = 1
+	spawn_positions = 1
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/den.dm:757]: Why did you cut shopkeeper down from 2 people to one?

File: code/modules/jobs/job_types/followers.dm [hunk 6]
```diff
@@ -242,8 +242,8 @@ Follower Volunteer
 	flag = F13FOLLOWERVOLUNTEER
 	department_flag = FOLLOWERS
 	faction = "Followers"
-	total_positions = 3
-	spawn_positions = 3
+	total_positions = 0
+	spawn_positions = 0
```
Comment (AdamElTablawy) [inline code/modules/jobs/job_types/followers.dm:246]: Why?

**Comment by AdamElTablawy** [comment]
Yeah, no. In its current state, this PR is definitely a 'don't fucking merge it.'

**Comment by Sinbest** [comment]
Farmer and prospector are now just citizens. Maybe you should have read the PR?

**Comment by BadAtThisGame302** [comment]
> Farmer and prospector are now just citizens. Maybe you should have read the PR?

Yeah but why, citizens won't do the farmers/prospectors jobs and those two jobs if not the prospector, were used

**Comment by John-Dang** [comment]
The old jobs have been added back, job counts are back to normal, as well as their loadouts.

**Comment by Mannybrado** [comment]
gonna go over the map and edit some shit, dont merge or anything right now please

**Comment by Mannybrado** [comment]
you should mark this as a draft right now by the way, weezer

**Comment by John-Dang** [comment]
> you should mark this as a draft right now by the way, weezer

Keeping edits to a second PR if this one passes.

**Comment by John-Dang** [comment]
In this state the PR should be ready for merge, all my edits are done. Anything past this point would be description edits.

**Comment by panzer1944** [comment]
honestly my main complaint with this PR is the followers- you've made them small, their entrance very fucky and you've put them out of the way. I'd suggest the following on first glance:
- Move the entry to the west-side
- Keep the ladder directly-connected to the building itself so we can still get the chad mining area (idk if we still have one, it looks very fucky and unorganized)
- Give us a second surgery set + table and shit, and expand the door into the surgery zone by just putting curtains instead of walls next to that one curtain
- Re-add the follower's little botany-bay
- Make it so you dont have to walk through the entire morgue to get up a ladder
- fix massive amount of wasted space at the cafeteria underground, give proper connections to it and the likes.
- give workshop, workbench and other shit - why do we not have it??

**Comment by John-Dang** [comment]
> honestly my main complaint with this PR is the followers- you've made them small, their entrance very fucky and you've put them out of the way. I'd suggest the following on first glance:
> 
> * Move the entry to the west-side
> * Keep the ladder directly-connected to the building itself so we can still get the chad mining area (idk if we still have one, it looks very fucky and unorganized)
> * Give us a second surgery set + table and shit, and expand the door into the surgery zone by just putting curtains instead of walls next to that one curtain
> * Re-add the follower's little botany-bay
> * Make it so you dont have to walk through the entire morgue to get up a ladder
> * fix massive amount of wasted space at the cafeteria underground, give proper connections to it and the likes.
> * give workshop, workbench and other shit - why do we not have it??

I agree with most of this except the mining area, botany bay, and workshop for the followers. Oasis has all those, they have a main mining dungeon map that can be used by anyone, it no longer requires prospector access only. And the farm in Oasis can be used by anyone. Same goes for the workshop. The followers are meant to be more part of Oasis in this update. The rest I can update though. Should take under an hour, and then it should be ready for push. Sound good?

**Comment by panzer1944** [comment]
> > honestly my main complaint with this PR is the followers- you've made them small, their entrance very fucky and you've put them out of the way. I'd suggest the following on first glance:
> > 
> > * Move the entry to the west-side
> > * Keep the ladder directly-connected to the building itself so we can still get the chad mining area (idk if we still have one, it looks very fucky and unorganized)
> > * Give us a second surgery set + table and shit, and expand the door into the surgery zone by just putting curtains instead of walls next to that one curtain
> > * Re-add the follower's little botany-bay
> > * Make it so you dont have to walk through the entire morgue to get up a ladder
> > * fix massive amount of wasted space at the cafeteria underground, give proper connections to it and the likes.
> > * give workshop, workbench and other shit - why do we not have it??
> 
> I agree with most of this except the mining area, botany bay, and workshop for the followers. Oasis has all those, they have a main mining dungeon map that can be used by anyone, it no longer requires prospector access only. And the farm in Oasis can be used by anyone. Same goes for the workshop. The followers are meant to be more part of Oasis in this update. The rest I can update though. Should take under an hour, and then it should be ready for push. Sound good?

Followers dont get access to the Oasis underground - they dont have citizen access, and the public shit always gets stolen and the followers tend to need something of their own so they can actually do shit without stealing shit from the wastes. You're trying to enforce the followers to use very public facilities, which enforces them being out of the already miniscule clinic that's out of the way, and thus decreases the downtime available for actually dealing with patients. You're not really making them more a part of oasis as you are forcing them to run all the way across the town they're not officially a part of so they can fulfill their basic needs like having enough meds to treat people.

**Comment by John-Dang** [comment]
> > > honestly my main complaint with this PR is the followers- you've made them small, their entrance very fucky and you've put them out of the way. I'd suggest the following on first glance:
> > > 
> > > * Move the entry to the west-side
> > > * Keep the ladder directly-connected to the building itself so we can still get the chad mining area (idk if we still have one, it looks very fucky and unorganized)
> > > * Give us a second surgery set + table and shit, and expand the door into the surgery zone by just putting curtains instead of walls next to that one curtain
> > > * Re-add the follower's little botany-bay
> > > * Make it so you dont have to walk through the entire morgue to get up a ladder
> > > * fix massive amount of wasted space at the cafeteria underground, give proper connections to it and the likes.
> > > * give workshop, workbench and other shit - why do we not have it??
> > 
> > 
> > I agree with most of this except the mining area, botany bay, and workshop for the followers. Oasis has all those, they have a main mining dungeon map that can be used by anyone, it no longer requires prospector access only. And the farm in Oasis can be used by anyone. Same goes for the workshop. The followers are meant to be more part of Oasis in this update. The rest I can update though. Should take under an hour, and then it should be ready for push. Sound good?
> 
> Followers dont get access to the Oasis underground - they dont have citizen access, and the public shit always gets stolen and the followers tend to need something of their own so they can actually do shit without stealing shit from the wastes. You're trying to enforce the followers to use very public facilities, which enforces them being out of the already miniscule clinic that's out of the way, and thus decreases the downtime available for actually dealing with patients. You're not really making them more a part of oasis as you are forcing them to run all the way across the town they're not officially a part of so they can fulfill their basic needs like having enough meds to treat people.

They can take donations from the community or trade with them, and citizen access is not needed to go in the Oasis underground now. Its part of the Oasis and visitable by anyone as it holds key things like the merchant, police, innhouse etc.

**Comment by panzer1944** [comment]
> They can take donations from the community or trade with them, and citizen access is not needed to go in the Oasis underground now. Its part of the Oasis and visitable by anyone as it holds key things like the merchant, police, innhouse etc.

Cant really trade if they cant reliably produce anything TO trade that wont just get stolen anyways due to how public your trying to make everything, and solely relying on donations is a genuinely horrible idea. The old clinic was fine as it was, just needed the skill-book spam removed in favour of just a chembook and surgerybook spare honestly, so why not just port it over and - you know - tilt it to fit?

**Comment by Keksk** [comment]
I like the idea of this PR! The buildings are made rather simplistic in comparison to the shopkeepers mansion.
However, I don't particularly like the layout.
I'd move most of the living space to the outer circle of the town and move buildings of importance further into the middle.
As for the followers, I'm simply going add a follow-up PR myself to get that straightened out.

A mere suggestion as for the layout. ( With no care for design and smoothing. Just the layout )




Either way I don't mind it too much. Change is always nice.

**Comment by John-Dang** [comment]
> I like the idea of this PR! The buildings are made rather simplistic in comparison to the shopkeepers mansion.
> However, I don't particularly like the layout.
> I'd move most of the living space to the outer circle of the town and move buildings of importance further into the middle.
> As for the followers, I'm simply going add a follow-up PR myself to get that straightened out.
> 
> A mere suggestion as for the layout. ( With no care for design and smoothing. Just the layout )
> 
> 
> 
> Either way I don't mind it too much. Change is always nice.

I'm planning to change the followers and some details about the town once the PR merges! Contact me on the discord and we can get into details on the plan. Personally I'm thinking of adding a designated open space for people to do little builds in.

**Comment by Singularity69** [comment]
shopkeeper door isn't locked

**Comment by BadIsMyUsername** [comment]
The store being underground is... awkward to say the least. As for Oasis itself, it is pretty cramped with no space to build anything extra, if you're the sorta guy who likes building custom places. I liked the idea in concept but actually having the chance to play it makes me want to never play any Oasis role ever again. 
P.S. Perhaps move the Mayoral/Citizen spawn point from a random house to what looks like the city hall. I spawned in and was greeted with a locked door and an angry citizen demanding how and why the Mayor had gotten into his still locked-up house.

**Comment by John-Dang** [comment]
> The store being underground is... awkward to say the least. As for Oasis itself, it is pretty cramped with no space to build anything extra, if you're the sorta guy who likes building custom places. I liked the idea in concept but actually having the chance to play it makes me want to never play any Oasis role ever again.
> P.S. Perhaps move the Mayoral/Citizen spawn point from a random house to what looks like the city hall. I spawned in and was greeted with a locked door and an angry citizen demanding how and why the Mayor had gotten into his still locked-up house.
> P.P.S why is the ladder for the Followers... in the church?

Alright so, I have another PR coming in to fix good issues brought up. I'm thinking of marking the mayor's house specially along with some other buildings. Think that's a good fix? Also yeah I should move em to the town hall perhaps. Aaand the church ladder's supposed to lead to the public soup kitchen.

**Comment by BadIsMyUsername** [comment]
Sounds good! Once again I'm not trying to shit on this out of malice, I like the shopkeeper's house design and Oasis was due for a makeover. Hope to see it all work out in the end.

**Pull request merged**